Makefile

```diff
@@ -1,13 +1,27 @@
 .PHONY: clean
-CFLAGS=-O2 -Wall -Wpedantic -Wextra -pthread
-LIBS=-lcrypt
+CFLAGS=-O2 -Wall -Wpedantic -Wextra -pthread -ggdb3
 OBJ=brute.o common.o main.o multi.o queue.o single.o
 TARGET=main
 
+ifeq ($(shell uname), Linux)
```

Comment:
I guess we could unconditionally link with our local version of crypt_r

Reply:
> I guess we could unconditionally link with our local version of crypt_r

Builded program on Podman machine, `make` output is
```
[core@localhost brute]$ make
cc -O2 -Wall -Wpedantic -Wextra -pthread -ggdb3   -c -o brute.o brute.c
cc -O2 -Wall -Wpedantic -Wextra -pthread -ggdb3   -c -o common.o common.c
cc -O2 -Wall -Wpedantic -Wextra -pthread -ggdb3   -c -o main.o main.c
cc -O2 -Wall -Wpedantic -Wextra -pthread -ggdb3   -c -o multi.o multi.c
cc -O2 -Wall -Wpedantic -Wextra -pthread -ggdb3   -c -o queue.o queue.c
cc -O2 -Wall -Wpedantic -Wextra -pthread -ggdb3   -c -o single.o single.c
cc -O2 -Wall -Wpedantic -Wextra -pthread -ggdb3 brute.o common.o main.o multi.o queue.o single.o -lcrypt -o main
```

Reply:
Debian is the same:
```
artemii@debian ~/C/brute (main)> make
cc -O2 -Wall -Wpedantic -Wextra -pthread   -c -o brute.o brute.c
cc -O2 -Wall -Wpedantic -Wextra -pthread   -c -o common.o common.c
cc -O2 -Wall -Wpedantic -Wextra -pthread   -c -o main.o main.c
cc -O2 -Wall -Wpedantic -Wextra -pthread   -c -o multi.o multi.c
cc -O2 -Wall -Wpedantic -Wextra -pthread   -c -o queue.o queue.c
cc -O2 -Wall -Wpedantic -Wextra -pthread   -c -o single.o single.c
cc -O2 -Wall -Wpedantic -Wextra -pthread brute.o common.o main.o multi.o queue.o single.o -lcrypt -o main
```

Reply:
Debian build has the same output:
```
artemii@debian ~/C/brute (multi-threaded-bruteforce)> make
cc -O2 -Wall -Wpedantic -Wextra -pthread -ggdb3   -c -o brute.o brute.c
cc -O2 -Wall -Wpedantic -Wextra -pthread -ggdb3   -c -o common.o common.c
cc -O2 -Wall -Wpedantic -Wextra -pthread -ggdb3   -c -o main.o main.c
cc -O2 -Wall -Wpedantic -Wextra -pthread -ggdb3   -c -o multi.o multi.c
cc -O2 -Wall -Wpedantic -Wextra -pthread -ggdb3   -c -o queue.o queue.c
cc -O2 -Wall -Wpedantic -Wextra -pthread -ggdb3   -c -o single.o single.c
cc -O2 -Wall -Wpedantic -Wextra -pthread -ggdb3 brute.o common.o main.o multi.o queue.o single.o -lcrypt -o main
```

Reply:
I mean - no need to do branching based on OS and always link with local copy of crypt_r.
Use ./crypt/*.[ha] for both Linux and MacOs.
	LIBS+=crypt/libcrypt.a
	CFLAGS+=-I./crypt